### PR TITLE
Feature/role group initialisation

### PR DESCRIPTION
**Pull request by mwnicks**
### What

Add check for role groups, adding them if missing, to api setup.

### How to review

Ensure all unit and feature tests are passing.
Ensure the service can be started and run correctly.

### Who can review

Anyone but me

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -132,3 +136,27 @@ func handleBodyUnmarshalError(ctx context.Context, err error) *models.ErrorRespo
...
+
+func initialiseRoleGroups(ctx context.Context, cognitoClient cognito.Client, userPoolId string) error {
+	adminGroup := models.NewAdminRoleGroup()
+	adminGroupCreateInput := adminGroup.BuildCreateGroupRequest(userPoolId)
+	_, err := cognitoClient.CreateGroup(adminGroupCreateInput)
+	if err != nil {
+		cognitoErr := models.NewCognitoError(ctx, err, "CreateGroup request for admin group from API start up")
+		if cognitoErr.Code != models.GroupExistsError {
+			return cognitoErr
+		}
```
Comment (daiLlew) [inline api/api.go:148]: Just making sure I follow this logic: If there is an error we only return it if its a Group exists error?

File: models/groups.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package models
+
+import "github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+
+const (
+	AdminRoleGroup     = "role-admin"
+	PublisherRoleGroup = "role-publisher"
+)
+
+type Group struct {
```
Comment (daiLlew) [inline models/groups.go:11]: Godoc on public types.

File: api/api.go (hunk 1 above)
Comment (mwnicks) [reply api/api.go:148]: No we return any errors that aren't `GroupExistsError`.
`GroupExistsError`s are swallowed.

The purpose of this function is to make sure the role groups are setup in Cognito, creating them if they aren't there.
If Cognito returns a success or a `GroupExistsError` then the group is in place and the service is ok to run. Any other error needs to be returned and stop the service starting.

File: models/groups.go (hunk 2 above)
Comment (mwnicks) [reply models/groups.go:11]: added Godocs for the type and functions

**Review by daiLlew — APPROVED** [review]

**Pull request merged**